gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -140,21 +138,26 @@ public void testByRoutingRulesEngineFileChange()
         assertEquals("etl",
                 routingGroupSelector.findRoutingGroup(mockRequest));
 
-        Thread.sleep(1);
-        // java.nio.file.attribute.FileTime offers a maximum precision of 1 ms. If the first
-        // half of this test runs in <1ms, the gateway may not recognize that the file
-        // has changed.
+        long etl2LastModified = etlLastModified;
+        while (!(etl2LastModified > etlLastModified)) {
```

Comment:
So now you are writing the file again and again until the lastModified shows a difference .. hmm .. just waiting and writing it once does not work?

Other thought ... could we fake the date of the first write to just be a bit older or does the OS/Java not allow that cheating.. I guess not but maybe..

Reply:
The `sleep(1)` wasn't fully effective at preventing test failures. I think the failure frequency decreased, but didn't resolve them completely as issue #123 shows. We could just sleep longer, but [given this note](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/File.html#lastModified())
>While the unit of time of the return value is milliseconds, the granularity of the value depends on the underlying file system and may be larger. For example, some file systems use time stamps in units of seconds.

this approached seemed safer.

Reply:
Not sure where your comment went off to... but you suggested setting lastUpdatedTime automatically, which is possible through the `nio` API. So this logic is now removed. Thanks!